### PR TITLE
Mcj 06

### PR DESCRIPTION
**Pull request by ThomassenCathleia**
The basic model structure of this project

### CODE REVIEW

File: src/be/mcjava/model/CookingOrders.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    public boolean isFinished() {
+        return finished;
+    }
+    
+    public void setFinished(boolean finished) {
+        this.finished = finished;
+    }
+    
+    public void addOrder(CustomerOrder order) {
+    }
```
Comment (lenshuygh) [inline src/be/mcjava/model/CookingOrders.java:63]: implementation:
ordersToCook.add(order);
?

File: src/be/mcjava/model/Ingredient.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    public int getCurrentStock() {
+        return currentStock;
+    }
+    
+    public void setCurrentStock(int currentStock) {
+        this.currentStock = currentStock;
+    }
+    
+    public boolean isAvailable() {
+        //todo: implement this, returns false so code will compile -Michiel 24/5/2019
```
Comment (lenshuygh) [inline src/be/mcjava/model/Ingredient.java]: implementation:
return (this.CurrentStock > 0);
?

File: src/be/mcjava/model/Product.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    public BigDecimal getPrice() {
+        return price;
+    }
+    
+    public void setPrice(BigDecimal price) {
+        this.price = price;
+    }
+    
+    public boolean isAvailable() {
+        //todo: implement this, returns false so code will compile -Michiel 24/5/2019
```
Comment (lenshuygh) [inline src/be/mcjava/model/Product.java]: implementation?
for (Ingredient ingredient : ingredients.keySet()) {
            if(ingredient.getCurrentStock() < ingredients.get(ingredient)){
                return false;
            }
        }
        return true;

**Comment by lenshuygh** [comment]
As with the other repo, very nice job, lot of work done already!!
added the same comments as the other repo to be complete, all about the helper-methods' implementations

**Comment by ThomassenCathleia** [comment]
Copy-pasting this from the other repo:

The main reason why I did not implement anything seriously was to just lay down the structure.

The boolean return methods are rather simple. The add/remove methods might need some better implementation, such as needing a custom equals and compare methods.

**Comment by lenshuygh** [comment]
copy/past also :-)

I get what you mean, but now they'll need to be implemented by making another story , if the methods are available and they don't do anything it will create confusion, especially since some do return something thats not a real result (eg: return false;)
If the method is available to other code you can't expect the person working on another story to go and check all your methods...
You can merge it as is but you'll need to add another story to make the implementations

File: src/be/mcjava/model/Product.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    public void setPrice(BigDecimal price) {
+        this.price = price;
+    }
+    
+    public boolean isAvailable() {
+        return ingredients.keySet()
+                .stream()
+                .allMatch(Ingredient::isAvailable);
+    }
+
```
Comment (lenshuygh) [inline src/be/mcjava/model/Product.java:59]: You only check if the stock is not null here, you should check if the stock is greater than or equal to the required amount

**Review by lenshuygh — CHANGES_REQUESTED** [review]
great work, only 1 thing to be fixed in Product.isAvailable()

File: src/be/mcjava/model/Product.java (hunk 4 above)
Comment (ThomassenCathleia) [reply src/be/mcjava/model/Product.java:59]: The method now also checks if the stock is greater or equal than the stock.

**Review by lenshuygh — APPROVED** [review]
nicely solved

**Review by Fonthom — APPROVED** [review]

**Pull request merged**